olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -790,6 +790,53 @@ bool VersionedLayerClientImpl::Release(const TileKeys& tiles) {
 
   return settings_.cache->Release(keys_to_release);
 }
+cache::KeyValueCache::KeyListType
+VersionedLayerClientImpl::GetRelatedPartitionKeys(
+    const std::string& partition_id) {
+  if (!settings_.cache) {
+    return {};
+  }
+
+  auto version = catalog_version_.load();
+  if (version == kInvalidVersion) {
+    OLP_SDK_LOG_WARNING(kLogTag,
+                        "Method Protect failed, version is not initialized");
+    return {};
+  }
+
+  auto cache = settings_.cache;
+
+  repository::PartitionsCacheRepository partitions_repo(catalog_, layer_id_,
+                                                        cache);
+  std::string handle;
+  if (partitions_repo.GetPartitionHandle(partition_id, version, handle)) {
+    repository::DataCacheRepository data_repo(catalog_, cache);
+
+    return cache::KeyValueCache::KeyListType{
+        partitions_repo.CreatePartitionKey(partition_id, version),
+        data_repo.CreateKey(layer_id_, handle)};
```

Comment:
Maybe instead of this, we should add the protect/release functionality to the repositories instead.